File: 001-100/007/nthPrime.py
from math import sqrt, ceil

# Check if a number is prime
def isPrime(n):
    for i in range(2, ceil(sqrt(int(n)))):
        if(int(n) % i == 0):
            return False
    return True
# Find the nth prime number
def nthPrime(n):
    i = 2; count = 0
    while(True):
        if(isPrime(i)):
            count += 1
            if(count == int(n)):
                return i
        i += 1

if __name__ == "__main__":
    # Get user input
    n = input("Enter an integer n: ")
    # Print results
    print("The " + "{:,}".format(int(n)) + "-th prime is")
    print("{:,}".format(nthPrime(n)))

File: 001-100/007/nthPrime.php
<?php
  // Check if a number is prime
  function isPrime($n){
    for ($i = 2; $i < $n; $i++) {
      if($n % $i == 0){ return False; }
    }
    return True;
  }
  // Find the nth prime number
  function nthPrime($n){
    $i = 2; $count = 0;
    while(True){
      if(isPrime($i)){
        $count++;
        if($count == $n){ return $i; }
      }
      $i++;
    }
  }
  // Get user input
  $prompt = "Enter an integer n: ";
  $n = readline($prompt);
  $nthPrime = nthPrime($n);
  // Print results
  echo "\nThe $n-th prime is";
  echo "\n" . $nthPrime . "\n";
?>
